Automattic-Tracks-iOS/Services/EventLoggingUploadQueue.swift

```diff
@@ -37,4 +43,23 @@ class EventLoggingUploadQueue {
             try fileManager.createDirectory(at: storageDirectory, withIntermediateDirectories: true, attributes: nil)
         }
     }
+
+    /// Ensure that logs don't pile up on the user's device if for some reason they can't be uploaded
+    func clean(retentionDays: Int) throws {
+
+        let date = Calendar.current.date(byAdding: .day, value: retentionDays * -1, to: Date())!
```

Comment:
True, though it should never be called more than once – this method is internal to the framework as is only called on the initial setup, so I'm ok keeping things clean/simple this way. If we rework this, it might be worth the extra complexity :)